source/configuration_manual/authentication/dict.rst

```diff
@@ -57,26 +71,56 @@ Dict configuration
     key = userdb/%u
     format = json
   }
+
+  # Space separated list of keys whose values contain key/value paired objects.
+  # All the key/value pairs inside the objects are added as passdb fields.
+  # This can only be used for JSON formatted values.
+  passdb_objects = passdb_domain_defaults passdb
+  # Userdb key/value object list.
+  userdb_objects = userdb
+
+With the above example configuration a passdb lookup for user@example.com
+would look like:
+
+ * Lookup passdb-domain/example.com
+ * Lookup passdb/user@example.com
+
+The results would have JSON-formatted strings, which would be merged. For
+example the results could be:
+
+ * passdb_domain_defaults/example.com: ``{ "proxy": "yes", "host": "backend.example.com" }``
+ * passdb/user@example.com: ``{ "password": "secret" }``
+ * userdb/user@example.com: ``{ "uid": 123, "gid": 345, "home": "/home/user@example.com" }``
+
+Individual value lookups
+========================
+
+Each %{dict:key} variable expansion does a dict lookup for the key. The value
+is used exactly as it is in the database without any kind of parsing.
+
+.. code-block:: none
+
+  key password {
+    key = passwords/%u
+    #format = value # value is the default
+  }
+  key proxy_host {
+    key = proxy-hosts/%u
+  }
+  passdb_fields {
+    proxy = yes
+    host = %{dict:proxy_host}
+    password = %{dict:password}
+  }
+
   key quota {
-    key = userdb/%u/quota # or e.g. quota/%{userdb:quota_class}
-    #format = value
+    key = quota/%{userdb:quota_class} # Assumes quota_class is already set
```

Comment:
Could the quota_class be looked up within this same dict already? Probably not? Verify & clarify that %{userdb:\*} / %{passdb:\*} aren't updated in the middle of this dict db handling, only after all the dict lookups are done.